### PR TITLE
sql: remove virtual index on information_schema.tables(table_name)

### LINKED ISSUE #55437: sql: v20.2 - information_schema.tables index on table_name only includes tables in the current search_path
**Describe the problem**

This bug affects CockroachDB version v20.2.0-beta.*.

When querying `information_schema.tables` for a specific table which resides in a custom schema (not in `public`) the 
string equality operator (`=`) does not work as expected on the `table_name` column. As a workaround one can use the
`like` operator with a wildcard (`%...%`) pattern which seems to work as expected.

**To Reproduce**

Spin up a CockroachDB cluster e.g. using `docker run --rm -it -p 26257:26257 cockroachdb/cockroach:latest-v20.2 start-single-node --insecure` and apply the following SQL statements.

```SQL
create schema nx;

create table nx.yy(a int);

select * from information_schema.tables where table_schema = 'nx' and table_name = 'yy'; 
--> unexpected empty result !!!

select * from information_schema.tables where table_name = 'yy'; 
--> unexpected empty result !!!

select * from information_schema.tables where table_schema = 'nx' and table_name like '%yy%' and length(table_name) = length('yy'); 
--> expected result with one row
```

**Environment:**
 - CockroachDB version `v20.2.0-beta.*`

### PR DESCRIPTION
**Pull request by rafiss**
fixes #55437 

This index has a bug because it only includes tables that are in the
current search_path. This means that a query that filters by equality on
table name would not include tables from user-defined schemas. The problem is
fixed for now by removing the index.

Release note (bug fix): The information_schema.tables metadata table no
longer ignores tables from other schemas when searching based on table
name.

### CODE REVIEW

File: pkg/sql/logictest/testdata/logic_test/information_schema [hunk 1]
```diff
@@ -517,6 +516,13 @@ query T
 SET DATABASE = test; SELECT table_name FROM information_schema.tables WHERE table_schema = 'other_db'
 ----
 
+# Check that tables from other schemas are found
+query TT rowsort
+CREATE SCHEMA myschema; CREATE TABLE myschema.abc(); CREATE TABLE abc(); SELECT table_schema, table_name FROM information_schema.tables WHERE table_name = 'abc'
```
Comment (mgartner) [inline pkg/sql/logictest/testdata/logic_test/information_schema:521]: Does this have to be 1 line?

**Review by mgartner — APPROVED** [review]
LGTM

**Review by RaduBerinde — COMMENTED** [review]
I assume we added this for a reason. Do we need to reopen any issue?

Release notes are not relative to 20.1, we did release various alphas/betas with this issue.

Code change LGTM

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/55472#-:-MJUG42CB_mqBcAwnzDZ:b-th6ic4)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan and @rafiss)

**Comment by rafiss** [comment]
> I assume we added this for a reason. Do we need to reopen any issue?

Don't think there was any particular reason. I think it was done as an example when virtual indexes were added in https://github.com/cockroachdb/cockroach/pull/47316/commits/2f7bc1c1015c869b5f7222c094a9f33e409e005b but I will ask @jordanlewis to confirm that.

> Release notes are not relative to 20.1, we did release various alphas/betas with this issue.

Got it, updating the release notes

File: pkg/sql/logictest/testdata/logic_test/information_schema (hunk 1 above)
Comment (rafiss) [reply pkg/sql/logictest/testdata/logic_test/information_schema:521]: looks like they don't -- i was just using the pattern in the test cases above. updated this example

**Comment by otan** [comment]
I think we expected a noice perf improvement with jordan's original change -- how hard is it to keep?

**Review by otan — APPROVED** [review]
yeah doesn't seem like there's a nice way of resolving this quickly.

**Comment by rafiss** [comment]
the perf improvement is more about the underlying infrastructure that jordan made to allow virtual tables to have "indexes" at all. from what I have seen and gathered, most tools and ORMs are more likely to query pg_catalog tables and benefit from the indexes in there (which still exist). do we know of any tools specifically that rely on `information_schema.tables` and search by `table_name`?

> how hard is it to keep?

seems tricky to me. the issue is that the code that this index uses to look up a table by name uses [`resolver.ResolveExistingObject` and `tree.ResolveExisting`](https://github.com/cockroachdb/cockroach/blob/03e8d3262a9a50d8ee12f822e4fbfb64bdf79cbf/pkg/sql/catalog/resolver/resolver.go#L153) which requires a schema name to be specified. so i guess one option is to change the index here to be on `information_schema.tables(schema_name, table_name)`. or is there a good way to get all of the `system.namespace` rows that match a given table name?

**Comment by rafiss** [comment]
actually, looks like virtual indexes can't be on multiple columns right now, so not doing the `(schema_name, table_name)`  index in this PR.

**Review by otan — APPROVED** [review]

**Pull request merged**